### ISSUE TITLE
D13: Does the local judge need to be 27B?

### ISSUE THREAD

**Issue by JiamanBettyWu** [JiamanBettyWu/retrieval-lab#2]
Migrated from `TODO.md` (2026-08-23). Keeps its D-number because `docs/plan.md`
and `SESSIONS.md` refer to it by that name.

**Context:** `Qwen3.8-27B` was picked on "bigger is safer" and never measured.
The principle worth testing: judging is *classification* against a rubric, and
everything it rules on is in the prompt — so parameter count may buy little.

**Unblocked 2026-08-23:** `mistral-small` (14GB) is pulled. Disk is not a
constraint; the mid-session 18GB reading was an artifact of the pull itself
(SESSIONS.md 2026-08-23).

**Options**
- **A)** Keep a 27B as recorded, unmeasured.
- **B)** Bake off 3-5 candidates against the 4c.1 fixture — it is small, so five
  cost about what one does.
- **C)** Drop the local judge, run Sonnet 5 alone.

**Recommendation:** B — decide on the fixture's evidence, not a spec sheet.

**Definition of done:** each candidate scored on the labelled fixture for
**separation** (does it distinguish grounded from ungrounded) *and*
**throughput**. The throughput number is what should choose the final n for the
full config sweep, rather than a placeholder.

**If C:** the "two judges" half of D10 is withdrawn and the kappa section loses
its agreement column.

**Blocked on:** labelled fixture rows (see `TODO.md`, "Pick up here").
Constraint from D10: the local judge must not be a qwen3 model — the generator
is `qwen3:8b`, and a shared pretraining corpus is exactly the self-bias to dodge.

**Comment by JiamanBettyWu** [comment]
**Resolved: B, and the answer to the title is no — but not for the reason the option assumed.**

Bake-off run 2026-08-28, `python -m retrieval_lab.judge --bakeoff`, rubric `v2`,
against the labelled seed-1 fixture (n=30: 16 answered, 14 refused).

## Separation

| judge | κ `grounded` | n | 95% CI | raw agree | parse-miss | s/call | digest |
|---|---|---|---|---|---|---|---|
| `mistral-small` (~24B) | **+0.586** | 16 | [+0.09, +1.00] | 13/16 (81%) | 0.0% | 42.45 | `8039dd90c113` |
| `gemma3:4b` | **−0.257** | 16 | [−0.67, +0.14] | 5/16 (31%) | 0.0% | 6.77 | `a2af6cc3eb7f` |

CIs are percentile bootstrap, 10,000 resamples. κ reproduced independently with
`sklearn.cohen_kappa_score` — matches the harness to 3dp.

Gap is 0.84, far outside the pre-registered tie band (<0.2 → tie, broken on
throughput). **The comparison resolves; it is not a tie, so throughput does not
get to decide.**

## The finding is not "bigger is better"

`gemma3:4b` parsed **100%** of rows and agreed **below chance**. Its errors are
systematic, not noisy — it ruled `false` on **8 of the 11** rows labelled `true`:

```
                   human→judge (grounded)
  gemma3:4b        TT 2 · FF 3 · TF 8 · FT 3
  mistral-small    TT 9 · FF 4 · FT 2 · TF 1
```

It has learned the *shape* of a strict groundedness ruling without the
discrimination: it fails nearly everything, including well-supported rationales.

This directly qualifies the principle in the issue body. "Everything it rules on
is in the prompt" is true, and it still does not follow that parameter count
buys little — the 4B model has the passages, the rationale and the rule in front
of it and cannot apply them. **What's in the prompt bounds what a judge could
know, not what it can do with it.**

Worth recording because it defeats the cheap health check: both candidates
passed `--smoke` 4/4 on format. A parse-rate monitor cannot distinguish these
two models. Only the labelled fixture can.

## Throughput → the n for the config sweep

At 42.45 s/call for the winner:

| n | per config | 4 configs + ceiling |
|---|---|---|
| 30 | 21 min | 1.8 h |
| 50 | 35 min | 2.9 h |
| **100** | **71 min** | **5.9 h** |

n=100 is affordable unattended (5.9 h overnight), so **throughput does not
constrain the sweep** — the label budget does. Recommend sizing on statistical
need, not on judge cost.

## Decision

- **Local judge: `mistral-small`, digest `8039dd90c113`.** Pinned by digest, not
  tag. Constraint from D10 respected — not a qwen3 model, so no shared-corpus
  self-bias with the `qwen3:8b` generator.
- `gemma3:4b` is disqualified on separation, not on format or speed.

## Caveats

- **n=16 on the certifying axis.** The CI `[+0.09, +1.00]` excludes zero and
  little else: this supports a *ranking*, not a value for κ. The ranking is
  secure (0.84 gap); the number is not.
- **Two candidates, not the 3–5 option B proposed.** This is a floor/ceiling
  probe across ~an order of magnitude in size, not a scaling curve. Where
  between 4B and 24B the capability appears is unmeasured.
- **`refusal_ok` is descriptive only** — `mistral-small` +0.632, `gemma3:4b`
  −0.105 (n=14). Two minority-class rows cannot certify anything; ~a third of
  bootstrap resamples contain no minority row and yield an undefined κ.
- Single annotator, so rater noise caps the achievable κ.

## Follow-ups this unblocks

- D10's "two judges or one" now has evidence and can be shaped into an issue:
  the second judge would be Sonnet 5 (needs an API branch in `judge_one`; the
  fixture, rubric and scoring are already transport-agnostic).
- Losing candidates can be `ollama rm`'d.

**Comment by JiamanBettyWu** [comment]
Closing: `mistral-small` (digest `8039dd90c113`) is the local judge, chosen on the fixture's evidence per option B. Result and reasoning in the comment above; recorded in `LEARNINGS.md` (2026-08-28) and now in `README.md`'s Phase 4 section.

Scope note worth keeping with the decision: the two candidates differ in vendor, architecture family and training corpus as well as parameter count (`gemma3:4b` Google 4.3B vs `mistral-small` Mistral 23.6B, both Q4_K_M). This is a **judge-selection** result, not a size result — the title's question is answered in the practical sense (no, a 27B judge was not needed) but the mechanism is unidentified. A size ablation would need two sizes within one family (`gemma3:4b` vs `gemma3:27b`) and was not run.

The remaining half of D10 — whether to add a second judge so their agreement becomes a reported result — stays in `TODO.md` until someone decides what that agreement column is *for*.
